week3/predict_citibike.Rmd

```diff
@@ -0,0 +1,249 @@
+# Citibike Data Prediction
+
+Author: Sambhav Shrestha
+
+
+## Importing the libraries
+
+```{r}
+library(tidyverse)
+library(scales)
+library(modelr)
+library(Holidays)
+
+```
+
+
+## Load the dataset
+```{r}
+trips <- read_tsv('trips_per_day.tsv')
+```
+
+## Data visualization
+```{r}
+# visualize the graph between each independent variable with num_trips to analyze the data
+trips %>%
+  mutate(holiday = isHoliday(ymd, "USFed")) %>%
+  mutate(wday = wday(ymd), month = month(ymd, label = T)) %>%
+  ggplot(aes(x = tmax, y = num_trips, color = factor(month))) +
+  geom_point()
+
+```
+
+## Splitting the data (Train and Test split)
+
+```{r}
+set.seed(5)
+
+num_days <- nrow(trips)
+train_frac <- 0.9
+num_train <- floor(num_days * train_frac)
+
+# sample certain fraction of random numbers between 1 and total rows of dataset
+sample_data <- sample(1:num_days, num_train, replace = FALSE)
+
+# training data (further needs to undergo cross validation)
+train <- trips[sample_data, ]
+
+# testing data
+test <- trips[-sample_data, ]
+
+```
+
+
+## Cross Validataion
+
+```{r}
+# no of k folds
+num_folds = 10
+
+# mutate the training dataset to include fold numbers
+train <- train %>%
+  mutate(fold = (row_number() %% num_folds) + 1)
+
+
+head(train)
+```
+
+## Model Testing for different degrees for each independent variable
+
+```{r}
+# testing the model for 8 polynomial degrees 
+N <- 1:8
+
+# create a vector for listing the errors
+avg_training_err = c()
+avg_validation_err = c()
+se_validation_err = c()
+
+# loop through each polynomial degrees
+for (n in N) {
+  
+  # do 10 fold cross validation within each 
+  validation_err = c()
+  train_err = c()
+  
+  for (f in 1:num_folds) {
+    
+    # fit on training data
+    train_data <- filter(train, fold != f)
+    model <- lm(num_trips ~ poly(tmin, n, raw = T) * poly(tmax, n), data = train_data)

```

Comment:
minor: could be helpful to have a comment with a plain language explanation of the model you are using